### DISCUSSION TITLE
Server- how do I trigger event from one namespace to other??

### DISCUSSION THREAD

**Discussion by rmControls** [miguelgrinberg/python-socketio#972] [Q&A]
I have a two namespace I want to trigger event on second namespace once the first event completed... (server side)

I tried like this but didnt worked....

(in simple words receive event in one namespace and emit to another namespace for another group of people...)

```python
class FistNameSpace(socketio.AsyncNamespace):
    async def on_connect(self, sid):
        pass
        #your connection code...

    async def on_first_action(self, sid, data):
        # your code
        result = some_function(data)
        SecondNameSpace().trigger_second_action(result)

class SecondNameSpace(socketio.AsyncNamespace):
    async def on_connect(self, sid):
        pass
        #your connection code...

    async def trigger_second_action(self, data):
        # your code
       await self.emit('my_event', data, room=myroom)
```

is this correct way to do it??

**Comment by miguelgrinberg** [comment]
This isn't a good way to think about it. The handlers that you have on your namespace classes are triggered by clients, they are not supposed to be triggered internally in the server. The second action that you need to happen should be a regular function or method, not an event handler.

**Reply by rmControls** [reply]
Yes, I understand...  but I have need to trigger second namespace event from server side... How to do it.? 
Ex: there are two different group of people connected to two different namespace based on their role... if person from first group trigger specific event then I want to broadcast some data to second group....

can you please advice correct way to achieve it??

**Reply by miguelgrinberg** [reply]
```python
def do_second_action(ns, data):
    await ns.emit('my_event', data, room=myroom)

class FistNameSpace(socketio.AsyncNamespace):
    async def on_connect(self, sid):
        pass
        #your connection code...

    async def on_first_action(self, sid, data):
        # your code
        result = some_function(data)
        do_second_action(self, result)

class SecondNameSpace(socketio.AsyncNamespace):
    async def on_connect(self, sid):
        pass
        #your connection code...

    async def trigger_second_action(self, data):
        # your code
       do_second_action(self, data)
```

**Reply by rmControls** [reply]
I am sorry I got confused with your example.... you called `do_second_action `function on both first and second namespace and no where called `trigger_second_action `event. if I assume `my_event` needs to replace with `trigger_second_action` at the top then it will becomes circular function...

also `on_first_action` sending self class to `do_second_action` function then how its been pointed to second namespace??

Sorry can you please explain a bit...

**Reply by miguelgrinberg** [reply]
Sorry, there was a small omission in my example:

```python
def do_second_action(ns, data):
    await ns.emit('my_event', data, room=myroom, namespace='/second-namespace')
```

It does not matter which namespace object you use, you can always specify the namespace explicitly.
